src/components/SwitchInput/SwitchInput.client.ts

```diff
@@ -0,0 +1,98 @@
+/**
+ * Based on https://github.com/argyleink/gui-challenges/blob/main/switch/index.js
```

Comment:
Is this a better alternative? https://devoorhoede.slack.com/archives/C02RU4J7H/p1709905150129279